mobile/lib/shared/providers/asset.provider.dart

```diff
@@ -92,14 +91,33 @@ class AssetNotifier extends StateNotifier<bool> {
     await _syncService.syncNewAssetToDb(newAsset);
   }
 
-  Future<void> deleteAssets(Iterable<Asset> deleteAssets) async {
+  Future<void> deleteAssets(
+    Iterable<Asset> deleteAssets, {
+    bool? force = false,
+  }) async {
     _deleteInProgress = true;
     state = true;
     try {
       final localDeleted = await _deleteLocalAssets(deleteAssets);
-      final remoteDeleted = await _deleteRemoteAssets(deleteAssets);
+      final remoteDeleted = await _deleteRemoteAssets(deleteAssets, force);
       if (localDeleted.isNotEmpty || remoteDeleted.isNotEmpty) {
-        final dbIds = deleteAssets.map((e) => e.id).toList();
+        // Local only assets are permanently deleted for now. So always remove them from db
+        var dbIds = deleteAssets
+            .where((a) => a.isLocal && !a.isRemote)
+            .map((e) => e.id)
+            .toList();
+        if (force == null || !force) {
+          final trashedAssets = remoteDeleted.map((e) {
+            e.isTrashed = true;
+            return e;
+          }).toList();
+          await _db.writeTxn(() async {
+            await _db.assets.putAll(trashedAssets);
```

Comment:
please move this DB action to the writeTxn below (where the deletes happen). You can use if statements in a writeTxn, e.g. if trashedAssets.isNotEmpty()

Reason: each writeTxn to the asset collection triggers a re-computation of the asset grid & and a UI rebuild. For users with large collections this takes a while. doing this twice in a row is an unpleasant user experience